github/GithubException.pyi

```diff
@@ -1,39 +1,30 @@
-from typing import Any, Dict, List, Optional, Tuple, Type, Union
+from typing import Any, Dict, Generic, List, Optional, Tuple, TypeVar, Union
 
 class GithubException(Exception):
-    def __init__(self, status: Union[int, str], data: Any, headers: Optional[Dict[str, str]]) -> None: ...
+    def __init__(
+        self, status: Union[int, str], data: Any, headers: Optional[Dict[str, str]]
+    ) -> None: ...
     def __str__(self) -> str: ...
     @property
     def data(self) -> Dict[str, Union[str, List[str], List[Dict[str, str]]]]: ...
     @property
     def status(self) -> int: ...
-
     @property
     def headers(self) -> Union[None, Dict[str, str]]: ...
 
-class BadAttributeException(GithubException):
+T = TypeVar("T")
+
+class BadAttributeException(GithubException, Generic[T]):
```

Comment:
Why does the class inherent from Generic[T]? I thought this class didn't wrap the attribute.

Reply:
The Generic is making the BadAttributeException be a generic class instead of a concrete class.  We cannot use the TypeVar inside the class without binding TypeVar T to the class.

So instead of having `BadAttributeException` we now have `BadAttributeException[T]`.

When initializing with `BadAttributeException(_, T, _)`, this now means `expected_value` will be typed as `expected_type(self) -> T`.

So if it is initialized with `str` for expected type, then `expected_value`'s type is `expected_type(self) -> str`.


This documentation should describe it better than I can.
https://mypy.readthedocs.io/en/stable/generics.html#defining-sub-classes-of-generic-classes